ppl/cmd/zqd/listen/command.go

```diff
@@ -141,7 +144,13 @@ func (c *Command) Run(args []string) error {
 	// Workers should registerWithRecruiter as late as possible,
 	// just before writing Port file for tests.
 	if c.conf.Personality == "worker" {
-		if err := core.WorkerRegistration(ctx, srv.Addr()); err != nil {
+		if _, _, err := net.SplitHostPort(c.conf.Recruiter); err != nil {
```

Comment:
Could you move these checks (the c.conf.Recruiter and c.conf.SpecNodeName) into WorkerRegistration?